### DISCUSSION TITLE
Converting from textmate doesn't work

### DISCUSSION THREAD

**Discussion by icecream17** [atom/atom#22324] [General]
The https://github.com/atom/language-typescript readme says that it's converted from https://github.com/microsoft/TypeScript-TmLanguage, which I locally cloned.

However if I try to convert any of the themes or grammars with the steps here: https://flight-manual.atom.io/hacking-atom/sections/converting-from-textmate/ it doesn't work.

Am I doing something wrong?

Edit: See https://github.com/atom/atom/issues/23159

**Comment by martinbira** [comment]
I can attest to this. Currently, the main reason for me not to switch from Sublime to Atom is that the conversion I'm trying to do of my favourite writing scheme that I've used for years cannot be converted. The file follows the syntax describe by [Sublime](https://www.sublimetext.com/docs/color_schemes_tmtheme.html) and I am using the exact command as specified by the [Atom Flight Manual]([https://flight-manual.atom.io/hacking-atom/sections/converting-from-textmate/#convert-the-theme).

The response I'm receiving is the following:
```
Could not find the required color settings in the theme.

The theme being converted must contain a settings array with all of the following keys:
  * background
  * caret
  * foreground
  * invisibles
  * lineHighlight
  * selection
```

and the first keys in the `settings` section is the following:
```
...
<dict>
    <key>name</key>
    <string>Writing Color Scheme</string>
    <key>colorSpaceName</key>
    <string>sRGB</string>
    <key>settings</key>
    <array>
        <dict>
            <key>settings</key>
            <dict>
                <key>background</key>
                <string>#F7F7F7</string>
                <key>caret</key>
                <string>#16BDEC</string>
                <key>foreground</key>
                <string>#181819</string>
                <key>invisibles</key>
                <string>#F7F7F7</string>
                <key>lineHighlight</key>
                <string>#00000008</string> 
                <key>selection</key>
                <string>#C5E8F4</string>
...
```
